automatic/360ts/update.ps1

```diff
@@ -1,25 +1,59 @@
 
 import-module au
 
-$releases = 'https://www.360totalsecurity.com/en/download-free-antivirus/360-total-security/?offline=1'
+function global:au_BeforeUpdate {
+  if ($Latest.Title -like '*essential*') {
+    cp "$PSScriptRoot\README_tse.md" "$PSScriptRoot\README.md" -Force
+  } else {
+    cp "$PSScriptRoot\README_ts.md" "$PSScriptRoot\README.md" -Force
+  }
+}
 
 function global:au_SearchReplace {
   @{
     ".\tools\chocolateyInstall.ps1" = @{
+      "(^\s*packageName\s*=\s*)('.*')" = "`$1'$($Latest.PackageName)'"
       "(^\s*url\s*=\s*)('.*')" = "`$1'$($Latest.URL32)'"
       "(^\s*checksum\s*=\s*)('.*')" = "`$1'$($Latest.Checksum32)'"
       "(^\s*checksumType\s*=\s*)('.*')" = "`$1'$($Latest.ChecksumType32)'"
+      "(^\s*softwareName\s*=\s*)('.*')" = "`$1'$($Latest.Title)'"
+    }
+    ".\360ts.nuspec" = @{
+      "(?i)(^\s*\<title\>).*(\<\/title\>)" = "`${1}$($Latest.Title)`${2}"
     }
   }
 }
 
-function global:au_GetLatest {
-  $HTML = Invoke-WebRequest -UseBasicParsing -Uri $releases
-  $url = $HTML.Links | ? href -match "\.exe$" | select -first 1 -expand href
-  $version = $url -split '_|(\.exe)' | select -last 1 -skip 2
+function Get360Version {
+param(
+ [string]$releases,
+	[string]$PackageName,
+	[string]$Title
+)
+$regex = '(\d+\.\d+\.\d+\.\d+)'
+
+$HTML =  Invoke-WebRequest -UseBasicParsing -Uri $releases | Foreach-Object { $_ -match $regex } | Out-Null
+$version = $Matches[0]
+$url = "https://free.360totalsecurity.com/totalsecurity/${PackageName}_Setup_${version}.exe"
+
+ @{
+    PackageName = $PackageName
```

Comment:
the packagename needs to be forced to either the original casing of the package, or to lowercase for new packages.

in this case it's enough to use `$PackageName.ToLower()`